### PR TITLE
Fix delay on simulcast uplink stream switches

### PR DESCRIPTION
**Pull request by hensmi-amazon**
**Issue #:** None

**Description of changes:** 

`updateLocalVideoFromPolicy` wasn't really working as expected for two reasons:
* `encodingSimulcastLayersDidChange` is not called until `chooseEncodingParameters` so it doesn't make sense to call before. Reversed the ordering here.
* `SimulcastUplinkObserver.encodingSimulcastLayersDidChange` is async so it would likely not be updated in time. Made this synchronous.
* `AttachMediaInputTask` and `ReceiveVideoInputTask` are run async (async on async!) so `chooseEncodingParameters` would also not be called in time.

**Testing:**
Smoke tested with three simulcast chrome senders with priority. Smoke tested with three non-simulcast chrome senders. Removed the third sender and there is no longer a long video freeze. There is a small one but that will not likely be fixable until we wire up the layers allocation extension again. This only occurs on participant number changes < 8 so its not very frequent.

*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
* 3 chrome senders using simulcast.
* One leaves conference
* No freezes on other applications

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
y

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
y

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
y

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/audiovideocontroller/DefaultAudioVideoController.test.ts [hunk 1]
```diff
@@ -1539,17 +1529,14 @@ describe('DefaultAudioVideoController', () => {
...
       audioVideoController.meetingSessionContext.transceiverController = new TestTransceiverController();
-      // @ts-ignore
-      audioVideoController.meetingSessionContext.signalingClient = new TestSignalingClient(
-        webSocketAdapter,
-        new NoOpDebugLogger()
-      );
 
+      // One that requires subscribe
+      audioVideoController.update({ needsRenegotiation: false });
+      // One that doesn't
```
Comment (ltrung) [inline test/audiovideocontroller/DefaultAudioVideoController.test.ts:1535]: this code confuses me :)

**Review by ltrung — APPROVED** [review]

File: test/audiovideocontroller/DefaultAudioVideoController.test.ts (hunk 1 above)
Comment (hensmi-amazon) [reply test/audiovideocontroller/DefaultAudioVideoController.test.ts:1535]: The unit testing war against async code (and components with many dependencies) will likely never be won. But this unit test was originally confirming the bug in the code :P ...

Comment (ltrung) [reply test/audiovideocontroller/DefaultAudioVideoController.test.ts:1535]: So the confusing part is `needsRenegotiation` is both set to false but the comment says one requires subcribe and one doesn't.  :D

Comment (hensmi-amazon) [reply test/audiovideocontroller/DefaultAudioVideoController.test.ts:1535]: `needsRenegotiation` could also have been named `forceRenegotiation` :)

**Pull request merged**